source/core/CEntryPoints.cpp

```diff
@@ -210,6 +210,29 @@ VIREO_EXPORT Int32 EggShell_GetArrayDimLength(TypeManagerRef tm, const char* viN
     return actualArray->GetLength(dim);
 }
 //------------------------------------------------------------
+//! Resizes a variable size Array symbol to have new dimension lengths specified by newLengths, it also initializes cells for non-flat data.
+//! Retuns -1 if the symbols is not found, -2 if was not possible to resize the array and 0 if resizing was successful.
+VIREO_EXPORT Int32 EggShell_ResizeArray(TypeManagerRef tm, const char* viName, const char* eltName, Int32 rank, Int32* newLengths)
+{
+    TypeManagerScope scope(tm);
+
+    SubString objectName(viName);
+    SubString path(eltName);
+    void *pData = null;
+
+    TypeRef actualType = tm->GetObjectElementAddressFromPath(&objectName, &path, &pData, true);
+    if (actualType == null || !actualType->IsArray()) {
+        return -1;
```

Comment:
Most of the other places are actually intended to be returning sizes, though, and are encoding -1 as an 'error' size.  The return value there is not a pure error code.